test/LockTest.js

```diff
@@ -1,74 +1,215 @@
+      assert.isTrue(hasInitialized)
     })
 
-    beforeEach(async () => {
-        await daoDeployment.deployBeforeEach(rootAccount)
-        const newLockAppReceipt = await daoDeployment.kernel.newAppInstance('0x1234', lockBase.address, '0x', false, {from: rootAccount})
-        lockForwarder = await Lock.at(deployedContract(newLockAppReceipt))
-        mockErc20 = await MockErc20.new(rootAccount, MOCK_TOKEN_BALANCE)
+    it('checks it is forwarder', async () => {
+      assert.isTrue(await lockForwarder.isForwarder())
     })
 
-    describe('initialize(address _token, uint256 _lockDuration, uint256 _lockAmount)', () => {
+    describe('changeLockDuration(uint256 _lockDuration)', () => {
+      it('sets a new lock duration', async () => {
+        await daoDeployment.acl.createPermission(rootAccount, lockForwarder.address, CHANGE_DURATION_ROLE, rootAccount)
+        const expectedLockDuration = 120
 
-        beforeEach(async () => {
-            lockForwarder.initialize(mockErc20.address, INITIAL_LOCK_DURATION, INITIAL_LOCK_AMOUNT)
-        })
+        await lockForwarder.changeLockDuration(expectedLockDuration)
+
+        const actualLockDuration = await lockForwarder.lockDuration()
+        assert.equal(actualLockDuration, expectedLockDuration)
+      })
+    })
 
-        it('sets variables as expected', async () => {
-            const actualToken = await lockForwarder.token()
-            const actualLockDuration = await lockForwarder.lockDuration()
-            const actualLockAmount = await lockForwarder.lockAmount()
-            const hasInitialized = await lockForwarder.hasInitialized()
+    describe('changeLockAmount(uint256 _lockAmount)', () => {
+      it('sets a new lock amount', async () => {
+        await daoDeployment.acl.createPermission(rootAccount, lockForwarder.address, CHANGE_AMOUNT_ROLE, rootAccount)
+        const expectedLockAmount = 20
 
-            assert.strictEqual(actualToken, mockErc20.address)
-            assert.strictEqual(actualLockDuration.toNumber(), INITIAL_LOCK_DURATION)
-            assert.strictEqual(actualLockAmount.toNumber(), INITIAL_LOCK_AMOUNT)
-            assert.isTrue(hasInitialized)
+        await lockForwarder.changeLockAmount(expectedLockAmount)
+
+        const actualLockAmount = await lockForwarder.lockAmount()
+        assert.equal(actualLockAmount, expectedLockAmount)
+      })
+    })
+
+    describe('forward(bytes _evmCallScript)', async () => {
+      let executionTarget, script
+      let addressLocks = 0
+
+      beforeEach(async () => {
+        //create script
+        executionTarget = await ExecutionTarget.new()
+        const action = {
+          to: executionTarget.address,
+          calldata: executionTarget.contract.execute.getData(),
+        }
+        script = encodeCallScript([action])
+      })
+
+      //should this test be done separately in say, 3 tests?
+      it('forwards action successfully', async () => {
```

Comment:
Typically the tests are structured using the AAA: arrange, act, assert pattern. So I would declare all the expected values at to the start of the function. In this case expectedNumberOfLocks and expectedBalance. Apply this pattern elsewhere if changing.

The size of this test is fine, no need for 3 tests.